src/main/java/com/synopsys/integration/alert/workflow/startup/component/CronJobsStartupComponent.java

```diff
@@ -0,0 +1,174 @@
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.workflow.startup.component;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.commons.lang3.math.NumberUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.exception.AlertDatabaseConstraintException;
+import com.synopsys.integration.alert.common.persistence.accessor.ConfigurationAccessor;
+import com.synopsys.integration.alert.common.persistence.model.ConfigurationFieldModel;
+import com.synopsys.integration.alert.common.persistence.model.ConfigurationModel;
+import com.synopsys.integration.alert.common.workflow.task.TaskManager;
+import com.synopsys.integration.alert.component.scheduling.SchedulingConfiguration;
+import com.synopsys.integration.alert.component.scheduling.descriptor.SchedulingDescriptor;
+import com.synopsys.integration.alert.workflow.scheduled.PhoneHomeTask;
+import com.synopsys.integration.alert.workflow.scheduled.PurgeTask;
+import com.synopsys.integration.alert.workflow.scheduled.frequency.DailyTask;
+import com.synopsys.integration.alert.workflow.scheduled.update.UpdateNotifierTask;
+
+@Component
+@Order(4)
+public class CronJobsStartupComponent extends StartupComponent {
+    private final Logger logger = LoggerFactory.getLogger(CronJobsStartupComponent.class);
+
+    private final ConfigurationAccessor configurationAccessor;
+    private final TaskManager taskManager;
+    private final PurgeTask purgeTask;
+    private final DailyTask dailyTask;
+    private final PhoneHomeTask phoneHomeTask;
+    private final UpdateNotifierTask updateNotifierTask;
+
+    @Autowired
+    public CronJobsStartupComponent(final ConfigurationAccessor configurationAccessor, final TaskManager taskManager, final PurgeTask purgeTask, final DailyTask dailyTask, final PhoneHomeTask phoneHomeTask,
```

Comment:
I think we should have a method on ScheduledTask along the lines of `initializeOnStartup()` or something. Then we can just autowire all of the ScheduledTasks and call that method; if they get scheduled some time after startup, the method could just do nothing.

Reply:
Since there is no order on this task then yes that would make sense to get all the tasks.  Filter the list of tasks to only the list where the initializeOnStartup returns true and then schedule them.